### DISCUSSION TITLE
Internet connection required?

### DISCUSSION THREAD

**Discussion by colinrblake** [creepymonster/GlucoseDirect#459] [General]
From what I understand, the Libre 3 apps reads the sensor and transfers the data to the LibreLinkUp app, and it's from there that GD can access the data. But how does the data get transferred from the Libre 3 app to the LibreLinkUp app? Is it:

1. All "on phone" via some kind of sharing
2. The Libre 3 app uploads the data to LibreView and from there the LIbreLinkUp app downloads it

My underlying question is this. If my phone doesn't have an internet connection (eg. I'm on an airplane), is GD going to have current sensor data?

**Comment by OddSquirrel** [comment]
That should be no. 2. The Libre 3 app uploads data to the cloud, where the LibreLinkUp app can grab it. In our case, Glucose Direct makes use of that same source as well. So, if your data won't land in the cloud, neither LibreLinkUp nor Glucose Direct can fetch it.

TL;DR: No internet connection, no data.

**Reply by colinrblake** [reply]
That's what I suspected. Thanks for the quick confirmation.

Will GD ever be able to read the sensor (Libre 3) directly?

**Reply by OddSquirrel** [reply]
You're most welcome. The problem is two-fold. Firstly, there is the problem of being able to decode the algorithm Abbott uses to dastardly deprive patients of their own data. This, of course, is being worked on by a few individuals. Secondly, there is the legal side. @creepymonster  may want to avoid trouble with Apple, who could easily nuke his app for violation of the law. As you probably know, it's either their way or the highway, so we're much more limited than we'd be on an Android device.

Who knows, maybe some European court will eventually come through for us. Keeping data from their rightful owners because of some copyright law is ethically questionable since it interferes with bettering disease management. Also, Apple might have to open their ecosystem to third parties soon, which would deliver us from having to use the current TestFlight workaround.
